src/Foundation/NSExpression.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Runtime.InteropServices;
+using XamCore.ObjCRuntime;
+
+namespace XamCore.Foundation {
+
+	public partial class NSExpression {
+
+		[Export ("arguments")]
+		public virtual NSExpression[] Arguments {
+			get {
+				var type = this.ExpressionType;
```

Comment:
style: we tend to avoid `this.` unless something (like locals/parameters) cause issues.